File: POS/POS/LogsForm.vb
Imports MySql.Data.MySqlClient

Public Class LogsForm
    Dim conn As New MySqlConnection
    Dim connstring As String = "Server=localhost;User Id=root;Password=;Database=pos"

    Private Sub LogsForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ComboBox1.SelectedIndex = 0
        ComboBox2.SelectedIndex = 0
        ComboBox3.SelectedIndex = 0

        refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text)
    End Sub

    Public Sub refreshdatagrid(searchfor As String, searchword As String, process As String, time As String)
        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim query As String = "SELECT * FROM logstbl WHERE "
            If searchfor = "Username" Then
                query = query & "username LIKE '%" & searchword & "%' "
            ElseIf searchfor = "Description" Then
                query = query & "description LIKE '%" & searchword & "%' "
            Else
                query = query & "username LIKE '%" & searchword & "%' "
            End If

            If process = "All" Then
                query = query
            Else
                query = query & "AND process = '" & process & "' "
            End If

            If time = "Today" Then
                query = query & "AND YEAR(dateandtime) = '" & Now.ToString("yyyy") & "' AND MONTH(dateandtime) = '" & Now.ToString("MM") & "' AND DAY(dateandtime) = '" & Now.ToString("dd") & "' ORDER BY dateandtime DESC"
            ElseIf time = "This Month" Then
                query = query & "AND YEAR(dateandtime) = '" & Now.ToString("yyyy") & "' AND MONTH(dateandtime) = '" & Now.ToString("MM") & "' ORDER BY dateandtime DESC"
            ElseIf time = "This Year" Then
                query = query & "AND YEAR(dateandtime) = '" & Now.ToString("yyyy") & "' ORDER BY dateandtime DESC"
            ElseIf time = "All" Then
                query = query & " ORDER BY dateandtime DESC"
            Else
                query = query & " ORDER BY dateandtime DESC"
            End If

            Dim comm As New MySqlCommand
            Dim reader As MySqlDataReader

            comm.Connection = conn
            comm.CommandText = query
            reader = comm.ExecuteReader

            DataGridView1.Rows.Clear()

            While reader.Read
                DataGridView1.Rows.Add(reader("username").ToString, reader("process").ToString, reader("description").ToString, reader("dateandtime").ToString)
            End While
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try
    End Sub

    Private Sub ComboBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox1.SelectedIndexChanged
        refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text)
    End Sub

    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TextBox1.TextChanged
        refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text)
    End Sub

    Private Sub ComboBox2_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox2.SelectedIndexChanged
        refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text)
    End Sub

    Private Sub ComboBox3_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox3.SelectedIndexChanged
        refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text)
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        MainForm.Show()
        Me.Close()
    End Sub

    Private Sub LogsForm_Closed(sender As Object, e As EventArgs) Handles Me.Closed
        MainForm.Show()
    End Sub
End Class

File: ACE Hardware/ACE Hardware/LogsForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace ACE_Hardware
{
    public partial class LogsForm : Form
    {

        private MySqlConnection conn = new MySqlConnection();
        private string connstring = Program.connstring;

        public LogsForm()
        {
            InitializeComponent();
        }

        private void LogsForm_Load(object sender, EventArgs e)
        {
            ComboBox1.SelectedIndex = 0;
            ComboBox2.SelectedIndex = 0;
            ComboBox3.SelectedIndex = 0;

            refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text);
        }


        public void refreshdatagrid(string searchfor, string searchword, string process, string time)
        {

                conn.ConnectionString = connstring;
                conn.Open();

                string query = "SELECT * FROM logstbl WHERE ";
                if (searchfor == "Username")
                    query = query + "username LIKE '%" + searchword + "%' ";
                else if (searchfor == "Description")
                    query = query + "description LIKE '%" + searchword + "%' ";
                else
                    query = query + "username LIKE '%" + searchword + "%' ";

                if (process == "All")
                    query = query;
                else
                    query = query + "AND process = '" + process + "' ";

                if (time == "Today")
                    query = query + "AND YEAR(dateandtime) = '" + DateTime.Now.ToString("yyyy") + "' AND MONTH(dateandtime) = '" + DateTime.Now.ToString("MM") + "' AND DAY(dateandtime) = '" + DateTime.Now.ToString("dd") + "' ORDER BY dateandtime DESC";
                else if (time == "This Month")
                    query = query + "AND YEAR(dateandtime) = '" + DateTime.Now.ToString("yyyy") + "' AND MONTH(dateandtime) = '" + DateTime.Now.ToString("MM") + "' ORDER BY dateandtime DESC";
                else if (time == "This Year")
                    query = query + "AND YEAR(dateandtime) = '" + DateTime.Now.ToString("yyyy") + "' ORDER BY dateandtime DESC";
                else if (time == "All")
                    query = query + " ORDER BY dateandtime DESC";
                else
                    query = query + " ORDER BY dateandtime DESC";

                MySqlCommand comm = new MySqlCommand();
                MySqlDataReader reader;

                comm.Connection = conn;
                comm.CommandText = query;
                reader = comm.ExecuteReader();

                DataGridView1.Rows.Clear();

                while (reader.Read())
                    DataGridView1.Rows.Add(reader["username"].ToString(), reader["process"].ToString(), reader["description"].ToString(), reader["dateandtime"].ToString());

                conn.Close();

        }

        private void ComboBox1_SelectedIndexChanged(object sender, EventArgs e)
        {
            refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text);
        }

        private void TextBox1_TextChanged(object sender, EventArgs e)
        {
            refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text);
        }

        private void ComboBox2_SelectedIndexChanged(object sender, EventArgs e)
        {
            refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text);
        }

        private void ComboBox3_SelectedIndexChanged(object sender, EventArgs e)
        {
            refreshdatagrid(ComboBox1.Text, TextBox1.Text, ComboBox2.Text, ComboBox3.Text);
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            Program.mainFormInstance.Show();
            this.Close();
        }

        private void LogsForm_FormClosed(object sender, FormClosedEventArgs e)
        {
            Program.mainFormInstance.Show();
        }
    }
}
